### DISCUSSION TITLE
LSP does not work with LSP installed inside Distrobox

### DISCUSSION THREAD

**Discussion by ydalton** [helix-editor/helix#10055] [Troubleshooting]
### Summary

Hi, and thank you for this wonderful editor! ✨

I found a problem with getting LSP to work. I don't get any LSP functionality, even though `hx --health` reports a working LSP config. For context, I install my LSPs in a distrobox container instead of my host, then export the binaries in my `~/.local/bin` directory:
```
pino@overbook-air:/tmp/cargotest/src$ hx --health rust
Configured language servers:
  ✓ rust-analyzer: /home/pino/.local/bin/rust-analyzer
Configured debug adapter: lldb-vscode
Binary for debug adapter: /home/pino/.local/bin/lldb-vscode
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```
This appears to be a bug because Emacs didn't have any problems with this arrangement, and I haven't tried LSP in Neovim at all so I don't if that works. Also, installing Helix in the distrobox and running it there does get LSP working, but I don't think it's an ideal solution. For the time being that is what I will do.

Running `hx --health rust` from the distrobox:
```
pino@my-distrobox:~$ hx --health rust
Configured language servers:
  ✓ rust-analyzer: /usr/bin/rust-analyzer
Configured debug adapter: lldb-vscode
Binary for debug adapter: /usr/bin/lldb-vscode
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

### Platform

Linux

### Terminal Emulator

kgx (GNOME Console)

### Installation Method

dnf

### Helix Version

helix 23.10 (f6021dd0)

**Comment by pascalkuthe** [comment]
Have you tried running `rust-analyzer --help` directly to figure out if it's possible to invoke this binary out side of distrobox? 

Also please provide a log file and use the latest release when reporting a bug

**Reply by ydalton** [reply]
`rust-analyzer` does indeed work...
```
pino@overbook-air:~$ rust-analyzer --help
rust-analyzer
  LSP server for the Rust programming language.

  Subcommands and their flags do not provide any stability guarantees and may be removed or
  changed without notice. Top-level flags that are not marked as [Unstable] provide
  backwards-compatibility and may be relied on.
...
```

Re logfile, sorry I did not provide it before. I will attach it here. A cursory glance at it shows that at least `clangd` (another LSP I tested) is being initialized, but nothing comes of it.
[helix.log](https://github.com/helix-editor/helix/files/14813371/helix.log)

**Comment by ydalton** [comment]
Okay, I tried configuring LSP with Neovim, and it has the same issue, so it's not specific to Helix. So I'll close the discussion.
